### ISSUE TITLE
RStudio crashes when running document()

### ISSUE THREAD

**Issue by FilipeamTeixeira** [hadley/devtools#1623]
Whenever I run devtools::document() RStudio crashes. However, when using the RStudio GUI (pressing the button), I get the following error:

```
Updating skynet documentation
Loading skynet

 *** caught segfault ***
address 0x1310, cause 'memory not mapped'

Traceback:
 1: dyn.load(file, DLLpath = DLLpath, ...)
 2: library.dynam(lib, package, package.lib)
 3: loadNamespace(package, ...)
 4: doTryCatch(return(expr), name, parentenv, handler)
 5: tryCatchOne(expr, names, parentenv, handlers[[1L]])
 6: tryCatchList(expr, classes, parentenv, handlers)
 7: tryCatch(loadNamespace(package, ...), error = function(e) e)
 8: requireNamespace(dep_name, quietly = TRUE)
 9: (function (dep_name, dep_ver = NA, dep_compare = NA) {    if (!requireNamespace(dep_name, quietly = TRUE)) {        stop("Dependency package ", dep_name, " not available.")    }    if (xor(is.na(dep_ver), is.na(dep_compare))) {        stop("dep_ver and dep_compare must be both NA or both non-NA")    }    else if (!is.na(dep_ver) && !is.na(dep_compare)) {        compare <- match.fun(dep_compare)        if (!compare(as.numeric_version(getNamespaceVersion(dep_name)),             as.numeric_version(dep_ver))) {            warning("Need ", dep_name, " ", dep_compare, " ",                 dep_ver, " but loaded version is ", getNamespaceVersion(dep_name))        }    }    return(TRUE)})(dots[[1L]][[1L]], dots[[2L]][[1L]], dots[[3L]][[1L]])
10: mapply(check_dep_version, deps$name, deps$version, deps$compare)
11: load_imports(pkg)
12: load_all(pkg)
13: devtools::document(roclets = c("rd", "collate", "namespace",     "vignette"))
14: withCallingHandlers(expr, packageStartupMessage = function(c) invokeRestart("muffleMessage"))
15: suppressPackageStartupMessages({    oldLC <- Sys.getlocale(category = "LC_COLLATE")    Sys.setlocale(category = "LC_COLLATE", locale = "C")    on.exit(Sys.setlocale(category = "LC_COLLATE", locale = oldLC))    devtools::document(roclets = c("rd", "collate", "namespace",         "vignette"))})
An irrecoverable exception occurred. R is aborting now ...

Exited with status 11.
```

If there's anything else you need from my side, let me know.

**Comment by jimhester** [comment]
Does this occur with any package or just a specific one? Have you run `devtools::clean_dll()` before running `devtools::document()`? Have you tried reinstalled roxygen2 and devtools?

**Comment by FilipeamTeixeira** [comment]
Tried everything you mentioned with no results.
This started after installing the new RStudio version. I started having random crashes, which didn't get solved by removing .rstudio-desktop and the usual files to reset everything. Even tried to reinstall R with no success.
As that was happening, I decided to tackle the issue package by package, until I got to devtools and data.table. Loading one of those results in R either crashing or giving a similar error to the one mentioned above.
